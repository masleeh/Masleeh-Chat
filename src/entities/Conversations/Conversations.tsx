import { List } from '@mui/material';
import { memo } from 'react';
import ConvItem from './elements/ConvItem';
import useDynamicReducer from 'shared/hooks/useDynamicReducer';
import { convReducer } from './model/slice/conversations.slice';

const Conversations = memo(() => {
    useDynamicReducer('coversation', convReducer, false)

    return (
        <List
            sx={{
                width: '100%',
                height: 'calc(100vh - 152px)',
                overflowY: 'scroll',
                scrollbarWidth: 'thin'
            }}
        >
            <ConvItem 
                username="masleeh"
                profile_pic={undefined}
                conv_id='12345'
                last_message={'hahaha'}
                isSelected={true}
                viewed={true}
            />
        </List>
    )
})

export default Conversations