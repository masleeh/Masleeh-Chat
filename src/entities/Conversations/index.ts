import { convActions, convReducer } from "./model/slice/conversations.slice";
import getConvsThunk from "./api/getConvsThunk/getConvsThunk";
import { getConvState } from "./model/selectors/getConvState/getConvState";
import { getConvData } from "./model/selectors/getConvData/getConvData";
import type {TConvItem} from './model/types/conversations.state'

export {
    convActions, 
    convReducer,
    getConvsThunk,
    getConvState,
    getConvData,
}

export type {
    TConvItem
}