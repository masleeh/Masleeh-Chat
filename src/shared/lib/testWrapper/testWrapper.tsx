import { render } from "@testing-library/react"
import { IStateSchema, TestStoreProvider } from "app/providers/store";
import i18nTest from 'shared/config/i18n/i18nTest'
import { I18nextProvider } from "react-i18next"
import { MemoryRouter } from "react-router-dom";
import { DeepPartial, EnhancedStore } from "@reduxjs/toolkit";
import { AppThemeProvider } from "app/providers/theme";

export interface componentRenderOptions {
    route?: string;
    initialState?: DeepPartial<IStateSchema>;
    store?: EnhancedStore
}

const testWrapper = (component: React.ReactNode, options: componentRenderOptions = {}) => {
    const {
        route = '/',
        initialState,
        store
    } = options

    return render(
        <TestStoreProvider initialState={initialState} store={store}>
            <AppThemeProvider>
                <MemoryRouter initialEntries={[route]}>
                    <I18nextProvider i18n={i18nTest}>
                        {component}
                    </I18nextProvider>
                </MemoryRouter>
            </AppThemeProvider>
        </TestStoreProvider>
    )
}

export default testWrapper