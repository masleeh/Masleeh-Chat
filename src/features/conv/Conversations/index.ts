import Conversations from "./ui/Conversation/Conversations";

export {
    Conversations
}