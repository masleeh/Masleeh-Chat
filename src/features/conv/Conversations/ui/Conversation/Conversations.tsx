import { List } from '@mui/material';
import { memo, useMemo } from 'react';
import ConvItem from '../ConvItem/ConvItem';
import useDynamicReducer from 'shared/hooks/useDynamicReducer';
import { convReducer } from 'entities/Conversations';
import useGetConversations from '../../model/useGetConversations/useGetConversations';
import ConvItemSkeleton from '../ConvItemSkeleton/ConvItemSkeleton';

const Conversations = memo(() => {
    useDynamicReducer('coversation', convReducer, false)

    const {
        isLoading,
        convData
    } = useGetConversations()

    const convItems = useMemo(() => {
        return convData
            // .filter((partData) => Boolean(partData.last_message))
            .map((partData) => (
                <ConvItem 
                    key={partData.conv_id}
                    partData={partData}
                />
            ))
    }, [convData])
    
    const loadingItems = useMemo(() => {
        const prevConvDataLength = Number(localStorage.getItem(
            import.meta.env.VITE_LOCALSTORAGE_CONVDATA_LENGTH
        ))

        return Array(prevConvDataLength > 5 ? prevConvDataLength : 5).fill(1).map((_, index) => (
            <ConvItemSkeleton key={index}/>
        ))
    }, [])

    return (
        <List
            sx={{
                width: '100%',
                height: 'calc(100vh - 152px)',
                overflowY: 'scroll',
                scrollbarWidth: 'thin'
            }}
        >
            {isLoading ? (
                loadingItems
            ) : (
                convItems   
            )}
        </List>
    )
})

export default Conversations