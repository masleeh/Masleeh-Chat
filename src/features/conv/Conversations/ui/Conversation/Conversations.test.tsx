import testWrapper from "shared/lib/testWrapper/testWrapper"
import Conversations from './Conversations'

describe('Conversations component', () => {
    it('Check Conversations', () => {
        const component = testWrapper(<Conversations />)
        expect(component).toMatchSnapshot()
    })
})