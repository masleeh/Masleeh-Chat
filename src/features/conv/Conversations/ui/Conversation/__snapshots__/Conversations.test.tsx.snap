// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Conversations component > Check Conversations 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding css-z6419k-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
              style="width: 40px; height: 40px;"
            />
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 150px;"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 250px;"
              />
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
              style="width: 40px; height: 40px;"
            />
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 150px;"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 250px;"
              />
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
              style="width: 40px; height: 40px;"
            />
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 150px;"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 250px;"
              />
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
              style="width: 40px; height: 40px;"
            />
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 150px;"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 250px;"
              />
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
              style="width: 40px; height: 40px;"
            />
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 150px;"
              />
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
                style="width: 250px;"
              />
            </p>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding css-z6419k-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
            style="width: 40px; height: 40px;"
          />
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 150px;"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 250px;"
            />
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
            style="width: 40px; height: 40px;"
          />
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 150px;"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 250px;"
            />
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
            style="width: 40px; height: 40px;"
          />
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 150px;"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 250px;"
            />
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
            style="width: 40px; height: 40px;"
          />
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 150px;"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 250px;"
            />
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <div
          class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-wkdwxb-MuiSkeleton-root"
            style="width: 40px; height: 40px;"
          />
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 150px;"
            />
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 250px;"
            />
          </p>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
